### PR TITLE
yomm2: new recipe

### PR DESCRIPTION
**Pull request by franramirez688**
Adding:  **yomm2/1.5.1**

Repository: https://github.com/jll63/yomm2

### CODE REVIEW

File: recipes/yomm2/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        if self.options.shared:
+            rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        else:  # header-only one
```
Comment (uilianries) [inline recipes/yomm2/all/conanfile.py:100]: Do you mean this project offer a header-only version? In that case, I would suggest adding `hearder_only` option in the recipe. The CI has a logic to process recipes that can be both library and header-only. The fmt is a good example: https://github.com/conan-io/conan-center-index/blob/master/recipes/fmt/all/conanfile.py#L23

Comment (RubenRBS) [reply recipes/yomm2/all/conanfile.py:100]: I agree with @uilianries, Cona can perfectly hanble the `header_only` option too and will set the package_type accordingly

Comment (franramirez688) [reply recipes/yomm2/all/conanfile.py:100]: Well, I do not think so. `fmt` offers that option and even creates a separate target if header-only, but `yomm2` does not do that. It creates the same target regardless of whether you are building a shared lib, but it behaves like a header-only or shared library. IMO, I think that adding that option could be confusing for consumers.

**Review by uilianries — DISMISSED** [review]
LGTM

**Review by uilianries — DISMISSED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/yomm2/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "YOMM2")
+        self.cpp_info.set_property("cmake_target_name", "YOMM2::yomm2")
+        if self.options.header_only:
+            self.cpp_info.bindirs = []
+            self.cpp_info.libdirs = []
+        else:  # shared-library
+            self.cpp_info.libs = ["yomm2"]
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                self.cpp_info.system_libs.append("m")
```
Comment (franramirez688) [inline recipes/yomm2/all/conanfile.py:116]:
```suggestion
        else:  # shared-library
            self.cpp_info.libs = ["yomm2"]
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.system_libs.append("m")
```

**Review by RubenRBS — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**